### PR TITLE
[flang][openacc][NFC] Get rid of use-desc-for-alloc=false option in atomic tests

### PR DESCRIPTION
**Pull request by clementval**
This option change the default lowering and should not be used. Make the atomic tests more homogenous to the rest of the OpenACC tests. No OpenACC tests use this option.

### CODE REVIEW

File: flang/test/Lower/OpenACC/acc-atomic-update.f90 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ! This test checks lowering of atomic and atomic update constructs
-! RUN: bbc --use-desc-for-alloc=false -fopenacc -emit-fir -hlfir=false %s -o - | FileCheck %s
-! RUN: %flang_fc1 -mllvm --use-desc-for-alloc=false -emit-fir -flang-deprecated-no-hlfir -fopenacc %s -o - | FileCheck %s
+! RUN: bbc -fopenacc -emit-fir -emit-fir -hlfir=false %s -o - | FileCheck %s
```
Comment (razvanlupusoru) [inline flang/test/Lower/OpenACC/acc-atomic-update.f90:2]: -emit-fir is here twice.

**Review by razvanlupusoru — APPROVED** [review]
LGTM

File: flang/test/Lower/OpenACC/acc-atomic-update.f90 (hunk 1 above)
Comment (clementval) [inline flang/test/Lower/OpenACC/acc-atomic-update.f90:2]:
```suggestion
! RUN: bbc -fopenacc -emit-fir -hlfir=false %s -o - | FileCheck %s
```

Comment (clementval) [reply flang/test/Lower/OpenACC/acc-atomic-update.f90]: Thanks for spotting this!

**Pull request merged**